### PR TITLE
MAT-7403:  Reworking the Saved Definitions list to use Lookup object …

### PR DESCRIPTION
**Pull request by gregory-akins**
…instead of string

## MADiE PR

Jira Ticket: [MAT-7403](https://jira.cms.gov/browse/MAT-7403)
(Optional) Related Tickets:

### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included
* [X] No extemporaneous files are included (i.e Complied files or testing results)
* [X] This PR is in to the **correct branch**.
* [X] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 1]
```diff
@@ -13,26 +13,31 @@ import {
...
-  cqlBuilderLookupsTypes: CqlBuilderLookupData | {};
+  cqlBuilderLookupsTypes: CqlBuilderLookup;
   textAreaRef;
   expressionEditorValue: string;
   setExpressionEditorValue: Function;
   setCursorPosition: Function;
   setAutoInsert: Function;
 }
 
+const localJunkData = ["Five", "Six,", "Seven", "Eight"];
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]: this is probably a debugging code

File: src/CqlBuilderPanel/definitionsSection/DefinitionsSection.test.tsx [hunk 2]
```diff
@@ -2,40 +2,107 @@ import * as React from "react";
 import { render, screen, waitFor, act } from "@testing-library/react";
 import DefinitionsSection from "./DefinitionsSection";
 import userEvent from "@testing-library/user-event";
-import { CqlBuilderLookupData } from "../../model/CqlBuilderLookup";
+import { CqlBuilderLookup, Lookup } from "../../model/CqlBuilderLookup";
 
 describe("DefinitionsSection", () => {
+  const tjcMeasurementPeriod = {
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/DefinitionsSection.test.tsx]: looks same as  `src/CqlBuilderPanel/__mocks__/MockCqlBuilderLookupsTypes.ts`. one of this can be deleted?

**Review by adongare — APPROVED** [review]
